substrate/frame/support/src/storage/migration.rs

```diff
@@ -385,6 +385,51 @@ pub fn move_prefix(from_prefix: &[u8], to_prefix: &[u8]) {
 	}
 }
 
+/// A phony migration that does nothing, except executing `TryDecodeEntireStorage` on
+/// `post_upgrade`, which implies it is only available if `try-state` feature is used.
+///
+/// This can be used typically in the top level runtime, where `AllPallets` typically comes from
+/// `construct_runtime!`.
+pub struct EnsureStateDecodes<AllPallets>(sp_std::marker::PhantomData<AllPallets>);
```

Comment:
Why is this a migration and not just part of `try-runtime`? I mean it already requires `try-runtime` and this could just be an integral part of it?

Reply:
This was discussed in the original PR description as a follow-up idea: https://github.com/paritytech/substrate/pull/13013#issue-1509379972. 

I still think it is good to add this reusable migration here so it can be added to any runtime ad-hoc, and later integrate it into `try-runtime-cli`.

Reply:
Makes no real sense what you are saying. That people can add this adhoc to their runtime, it requires them to update their deps. If instead of adding this as an extra migration, we directly add it to the try runtime logic. People will need to update the deps any way to get the changes.

This can also not really be integrated into try runtime cli, if you are not planing on writing something with subxt that does this.

Reply:
We can just call the migration from `Executive::try_runtime_upgrade` if we want to bake it into the system permanently.  
It gets passed a `UpgradeCheckSelect` that we could use to select whether it should run.

Reply:
I'm in favour of this being configurable via `UpgradeCheckSelect`, like the try-state checks. Or, maybe just run it if the user is running `try-state` checks. Since a really obvious storage invarient is that it should be decodable.

Reply:
> when `UpgradeCheckSelect` is configured to run any `try-state` checks. It seems a really obvious storage invariant that all storage should be decodable.

Sounds good!

Reply:
done